src/main/java/org/capstone/job_fair/services/impl/job_fair/JobFairServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package org.capstone.job_fair.services.impl.job_fair;
+
+import org.capstone.job_fair.config.jwt.details.UserDetailsImpl;
+import org.capstone.job_fair.constants.DataConstraint;
+import org.capstone.job_fair.constants.MessageConstant;
+import org.capstone.job_fair.models.dtos.job_fair.JobFairDTO;
+import org.capstone.job_fair.models.entities.account.AccountEntity;
+import org.capstone.job_fair.models.entities.job_fair.JobFairEntity;
+import org.capstone.job_fair.models.enums.Role;
+import org.capstone.job_fair.models.statuses.JobFairStatus;
+import org.capstone.job_fair.repositories.account.AccountRepository;
+import org.capstone.job_fair.repositories.job_fair.JobFairRepository;
+import org.capstone.job_fair.services.interfaces.job_fair.JobFairService;
+import org.capstone.job_fair.services.mappers.JobFairMapper;
+import org.capstone.job_fair.utils.MessageUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class JobFairServiceImpl implements JobFairService {
+    @Autowired
+    JobFairRepository jobFairRepository;
+    @Autowired
+    AccountRepository accountRepository;
+    @Autowired
+    private JobFairMapper jobFairMapper;
+
+    @Override
+    public void createJobFair(JobFairDTO dto) {
```

Comment:
please add more time constraint 
Job fair's state: companyRegisterStartTime -> companyRegisterEndTime -> companyBuyBoothStartTime -> companyBuyBoothEndTime -> attendantRegisterStartTime  -> startTime -> endTime
Each of these states has an interval between them.
Ex companyRegisterEndTime - companyRegisterStartTime < 1 week
These time constraints is not yet defined so you can use any time.